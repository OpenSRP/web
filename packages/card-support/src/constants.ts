// OPENSRP API endpoints
export const OPENSRP_URL_USER_ASSIGNMENT = '/organization/user-assignment';
export const OPENSRP_URL_CLIENT_SEARCH = '/client/search';
