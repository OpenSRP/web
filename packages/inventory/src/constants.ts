// opensrp api strings
export const OPENSRP_API_BASE_URL = 'https://mg-eusm-staging.smartregister.org/opensrp/rest/';

// router routes
export const INVENTORY_SERVICE_POINT_LIST_VIEW = '/inventory/list';
export const INVENTORY_SERVICE_POINT_PROFILE_VIEW = '/inventory/list';

// other constants
/** namespace for the keys attached to the columns */
export const TableColumnsNamespace = 'inventory';

// magic strings
export const SEARCH_QUERY_PARAM = 'querySearch';
export const TABLE_PAGE_SIZE = 5;
export const TABLE_PAGE_SIZE_OPTIONS = ['5', '10', '20', '50', '100'];

export const tablePaginationOptions = {
  showQuickJumper: true,
  showSizeChanger: true,
  defaultPageSize: TABLE_PAGE_SIZE,
  pageSizeOptions: TABLE_PAGE_SIZE_OPTIONS,
};

//endpoints
export const GET_INVENTORY_BY_SERVICE_POINT = 'stockresource/servicePointId/';

//payload
export const inventory1 = {
  stockId: '5dfcaecd-430a-4284-972f-9acf87775a9b1',
  productName: 'Scale1',
  unicefSection: 'Health1',
  quantity: 4,
  deliveryDate: '2020-02-01',
  accountabilityEndDate: '2020-10-10',
  donor: 'ADB',
  servicePointId: '903971',
  poNumber: 1111,
  serialNumber: '1234serial2',
  providerId: 'providerId',
};

export const inventory2 = {
  stockId: '5dfcaecd-430a-4284-972f-9acf87775a9b2',
  productName: 'Scale2',
  unicefSection: 'Health2',
  quantity: 4,
  deliveryDate: '2020-02-01',
  accountabilityEndDate: '2020-10-10',
  donor: 'ADB',
  servicePointId: '903972',
  poNumber: 1112,
  serialNumber: '1234serial2',
  providerId: 'providerId',
};
