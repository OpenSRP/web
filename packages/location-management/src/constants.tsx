// URLs
export const URL_ALL_LOCATION_TAGS = '/location-tag';
export const URL_DELETE_LOCATION_TAGS = '/location-tag/delete';
export const KEYCLOAK_API_BASE_URL ='https://opensrp-stage.smartregister.org/opensrp/rest';
