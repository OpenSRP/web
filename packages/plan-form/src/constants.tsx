export const A1 = 'A1';
export const A2 = 'A2';
export const B1 = 'B1';
export const B2 = 'B2';
export const LOW_PRIORITY = 'low-priority';
export const MEDIUM_PRIORITY = 'medium-priority';
export const HIGH_PRIORITIY = 'high-priority';
export const TRUE = 'True';
export const FALSE = 'False';
export const DISABLED = 'Disabled';
export const INTERNAL = 'internal';
export const IGNORE = 'ignore';
export const INTERVENTION_TYPE_CODE = 'interventionType';
export const FI_STATUS_CODE = 'fiStatus';
export const FI_REASON_CODE = 'fiReason';
export const OPENSRP_EVENT_ID_CODE = 'opensrpEventId';
export const CASE_NUMBER_CODE = 'caseNum';
export const TASK_GENERATION_STATUS_CODE = 'taskGenerationStatus';
export const TEAM_ASSIGNMENT_STATUS_CODE = 'teamAssignmentStatus';
export const CASE_TRIGGERED = 'Case Triggered';
export const ROUTINE = 'Routine';
export const INVESTIGATION = 'Investigation';
export const DAYS = 'days';
export const CONDITION = 'condition';
export const TRIGGER = 'trigger';

// plan activity codes
export const CASE_CONFIRMATION_ACTIVITY_CODE = 'caseConfirmation';
export const FAMILY_REGISTRATION_ACTIVITY_CODE = 'familyRegistration';
// opensrp api strings
export const PLAN_LIST_URL = '/plans/list';
export const OPENSRP_API_BASE_URL = '';
export const OPENSRP_PLAN_ENDPOINT = 'plans';