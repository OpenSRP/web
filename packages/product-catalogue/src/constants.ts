// opensrp api strings
export const OPENSRP_PRODUCT_CATALOGUE = 'rest/product-catalogue';

// router routes
export const CATALOGUE_LIST_VIEW_URL = '/product-catalogue';
export const CATALOGUE_EDIT_VIEW_URL = '/product-catalogue/edit';
export const CATALOGUE_CREATE_VIEW_URL = 'product-catalogue/new';
export const HOME_URL = '/';

// other constants
/** namespace for the keys attached to the columns */
export const TableColumnsNamespace = 'product-catalogue';

/** route params for product-catalogue pages */
export interface RouteParams {
  productId?: string;
}
