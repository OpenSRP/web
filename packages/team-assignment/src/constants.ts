export const OPENSRP_API_BASE_URL = 'https://opensrp-staging.smartregister.org/opensrp/rest/';

// router routes
export const TEAM_ASSIGNMENT_LIST_VIEW_URL = '/team-assignment';
export const TEAM_ASSIGNMENT_EDIT_VIEW_URL = '/team-assignment/edit';
export const TEAM_ASSIGNMENT_CREATE_VIEW_URL = 'team-assignment/new';
export const HOME_URL = '/';
export const LOCATION_HIERARCHY_ENDPOINT = 'location/hierarchy';
export const PLANS_ENDPOINT = 'plans';
export const ASSIGNMENTS_ENDPOINT = 'organization/assignLocationsAndPlans';
export const SUCCESSFULLY_ASSIGNED_TEAMS = 'Successfully Assigned Teams';
export const ENTER_TEAM_NAME = 'Enter a Team name';

// other constants
/** namespace for the keys attached to the columns */
export const TableColumnsNamespace = 'team-assignment';

// Table view strings

export const TEAM_ASSIGNMENT_PAGE_TITLE = 'Team Assignment';
export const SAVE = 'Save';
export const CANCEL = 'Cancel';
