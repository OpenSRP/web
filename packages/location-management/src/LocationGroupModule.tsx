import React from 'react';
import { Route, Switch } from 'react-router';
import { BrowserRouter } from 'react-router-dom';

import { URL_LOCATION_GROUP_ADD, URL_LOCATION_GROUP_EDIT } from './constants';
import LocationGroupView from './components/LocationGroupView';
import LocationGroupAdd from './components/LocationGroupAdd';

/**
 * Location Group Module
 *
 * @returns {Element} - Location Group Module
 */
export const LocationGroupModule: React.FC = () => {
  return (
    <BrowserRouter>
      <Switch>
        <Route path={URL_LOCATION_GROUP_EDIT + '/:id'} component={LocationGroupAdd} />
        <Route path={URL_LOCATION_GROUP_ADD} component={LocationGroupAdd} />
        <Route component={LocationGroupView} />
      </Switch>
    </BrowserRouter>
  );
};

export default LocationGroupModule;
