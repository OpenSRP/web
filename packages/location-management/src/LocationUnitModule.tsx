import React from 'react';
import { Route, Switch } from 'react-router';
import { BrowserRouter } from 'react-router-dom';
import { URL_LOCATION_UNIT_ADD } from './constants';

import ConnectedLocationUnitView from './components/LocationUnitView';
import LocationUnitAdd from './components/LocationUnitAdd';

/**
 * Location Unit Module
 *
 * @returns {Element} - Location Unit Module
 */
export const LocationUnitModule: React.FC = () => {
  return (
    <BrowserRouter>
      <Switch>
        <Route exact component={LocationUnitAdd} />
        <Route exact component={ConnectedLocationUnitView} />
      </Switch>
    </BrowserRouter>
  );
};

export default LocationUnitModule;
