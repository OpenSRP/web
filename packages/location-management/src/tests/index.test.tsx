import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import { store } from '@opensrp/store';

import { LocationUnitModule } from '../LocationUnitModule';
import { LocationGroupModule } from '../LocationGroupModule';
import { URL_LOCATION_TAG, URL_LOCATION_UNIT } from '../constants';

describe('containers/pages/locations/LocationGroupAddition', () => {
  it('renders', () => {
    mount(
      <MemoryRouter initialEntries={[URL_LOCATION_TAG]}>
        <Provider store={store}>
          <LocationGroupModule />
        </Provider>
      </MemoryRouter>
    );
  });
});

describe('containers/pages/locations/LocationUnitModule', () => {
  it('renders', () => {
    mount(
      <MemoryRouter initialEntries={[URL_LOCATION_UNIT]}>
        <Provider store={store}>
          <LocationUnitModule />
        </Provider>
      </MemoryRouter>
    );
  });
});
