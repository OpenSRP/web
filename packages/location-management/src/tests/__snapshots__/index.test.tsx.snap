// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/pages/locations/LocationTagAddition renders without crashing 1`] = `
Object {
  "children": <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <LocationTagModule />
  </Provider>,
  "initialEntries": Array [
    "/location/group",
  ],
}
`;
